src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,217 @@
+
+  useEffect(() => {
+    setSelectedItems(items)
+  }, [items, setSelectedItems])
+
+  const setSingleItem = (item: Item) => {
+    for (const i of matchingItems) {
+      if (i.id !== item.id) {
+        deselectItem(i)
+      }
+    }
+    selectItem(item)
+  }
+
+  const toggleItem = (item: Item) => {
+    const isSelected = selectedItems.some(
+      (selectedItem: Item) => selectedItem.id === item.id
+    )
+
+    if (isSingleSelectable) {
+      setSingleItem(item)
+    } else if (isSelected) {
+      deselectItem(item)
+    } else {
+      selectItem(item)
+    }
+  }
+
+  const handleOnChange = (event: React.FormEvent<HTMLInputElement>) => {
+    setSuggestionIndex(-1)
+    setSearchTerm(event.currentTarget.value)
+  }
+
+  const selectItem = (selectedItem: Item) => {
+    setItems((previousState) => [...new Set([...previousState, selectedItem])])
+  }
+
+  const deselectItem = (selectedItem: Item) =>
+    setItems((previousState) =>
+      previousState.filter((item) => item.id !== selectedItem.id)
+    )
+
+  const resultsToShow = Math.min(matchingItems.length, maxAmountOfItems)
+
+  const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = (e) => {
+    if (e.key === 'Escape' || e.key === 'Esc' || e.key === 'Tab') {
+      setDoShowResultsList(false)
+      setSuggestionIndex(-1)
+    }
+    if (e.key === 'Enter') {
+      e.preventDefault()
+      const item = matchingItems[suggestionIndex]
+      if (item) {
+        toggleItem(item)
+      }
+    }
+    if (e.key === 'ArrowDown' || e.key === 'Down') {
+      e.preventDefault()
+      setDoShowResultsList(true)
+      if (suggestionIndex === -1) {
+        setSuggestionIndex(0)
+        return
+      }
+
+      const isGreaterThanResultsListLength = suggestionIndex < resultsToShow - 1
+      setSuggestionIndex(
+        isGreaterThanResultsListLength ? suggestionIndex + 1 : 0
+      )
+    }
+    if (e.key === 'ArrowUp' || e.key === 'Up') {
+      e.preventDefault()
+      setDoShowResultsList(true)
+      if (suggestionIndex === -1) {
+        setSuggestionIndex(0)
+        return
+      }
+      const isLessThanResultsListLength = suggestionIndex <= 0
+      setSuggestionIndex(
+        isLessThanResultsListLength ? resultsToShow - 1 : suggestionIndex - 1
+      )
+    }
+  }
+
+  return (
+    <Styled.Wrapper>
+      <Styled.DropdownWrapper ref={wrapperRef}>
+        <Styled.Label>{label}</Styled.Label>
+        <Styled.InputWrapper>
+          <Styled.DropdownInput
+            aria-label={`Change ${label} ${
+              isSingleSelectable ? 'option' : 'options'
+            }`}
+            placeholder={placeholder}
+            onChange={handleOnChange}
+            onKeyDown={handleKeyDown}
+            value={enableSearch ? undefined : placeholder}
+            onClick={() => {
+              setDoShowResultsList(!doShowResultsList)
+            }}
+            aria-controls={`${id} ${controls}`}
+            aria-describedby={`${id}-help`}
```

Comment:
Why -help?

Reply:
Well they need different ID:s so a screen reader or other assistive technology doesn't get confused. This part is just for adding supplemental information which will basically add "You have an autocomplete with options available below the input field".